drivers/video/video_stm32_dcmi.c

```diff
@@ -244,16 +246,16 @@ static int video_stm32_dcmi_stream_start(const struct device *dev)
 {
 	struct video_stm32_dcmi_data *data = dev->data;
 	const struct video_stm32_dcmi_config *config = dev->config;
-	size_t buffer_size = data->pitch * data->height;
 
-	data->buffer = k_heap_alloc(&video_stm32_buffer_pool, buffer_size, K_NO_WAIT);
-	if (data->buffer == NULL) {
-		LOG_ERR("Failed to allocate DCMI buffer for image. Size %d bytes", buffer_size);
+	data->vbuf = k_fifo_get(&data->fifo_in, K_NO_WAIT);
+
+	if (data->vbuf == NULL) {
+		LOG_ERR("Failed to dequeue a DCMI buffer.");
```

Comment:
After this commit, if a video buffer is not passed before start(), this would return an error.
I think that all examples in `main` would still work:

here: 2 buffers
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/video/tcpserversink/src/main.c#L41
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/video/capture_to_lvgl/src/main.c#L22

or otherwise, `CONFIG_VIDEO_BUFFER_POOL_NUM_MAX`
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/video/capture/src/main.c#L85

Reply:
Yes, the applications should still work, mine still works without changes:

App: allocates 2 buffers.
App: enqueues 2 buffers in fifo_in.
App: starts streaming.
Driver: dequeues buffer 1 from fifo_in and holds it for use as DCMI buffer.
Driver: Frame ready, dequeue buffer 2 from fifo_in, memcpy, enqueue in fifo_out.